### ISSUE TITLE
Attribute behaves differently than annotation in case of DocBlockDescriptions

### ISSUE THREAD

**Issue by kminek** [zircote/swagger-php#1668]
This works:

```
@OA\Schema(type="string", description=null)
```

this dont:

```
`#[OA\Schema(type: 'string', description:null)]`
```

**Comment by DerManoMann** [comment]
A bit more detail might be good. 'doesn't work's could be anything...

Best guess is you are facing this issue: https://zircote.github.io/swagger-php/guide/faq.html#warning-required-oa-info-not-found

**Comment by kminek** [comment]
@DerManoMann right. here is my use case. i have a enum class annotated with attribute:

```

/**
 * Some docblock that should not be exposed in API docs.
 */
#[OA\Schema(type: 'string')]
enum Language: string
{
...
}
```

When API docs are generated, also enum class docblock is parsed and included in generated docs. However as this docblock contains internal info which should not be exposed i tried to set `description` to `null`:

```
/**
 * Some docblock that should not be exposed in API docs.
 */
#[OA\Schema(type: 'string', description: null)]
enum Language: string
{
...
}
```

But still the docblock is included in API docs.

**Comment by DerManoMann** [comment]
Ah, yes, sorry. Was reading this on my phone and didn't scroll right :/

The attributes do not use `Generator::UNDEFINED`, so null is not really considered a valid value. Have you tried an empty string? Not sure of the top of my head how the code evaluates whether a string is set or not.
I suppose an ugly 'fix' would be to set the value to `Generator::UNDEFINED`, although that does not look very nice :{

**Comment by DerManoMann** [comment]
I wonder if for attributes all properties typed `?string` could treat empty string the same as `null`... hmmm

**Comment by kminek** [comment]
> I wonder if for attributes all properties typed `?string` could treat empty string the same as `null`... hmmm

dont really know the internals of this library but as a library consumer i would expect same behavior no matter what "annotating interface" i am using

**Comment by DerManoMann** [comment]
Fair enough - will do some more digging then ;)

**Comment by DerManoMann** [comment]
Fixed via #1886
